include/arch/common/sys_bitops.h

```diff
@@ -42,6 +42,20 @@ static ALWAYS_INLINE int sys_test_bit(mem_addr_t addr, unsigned int bit)
 	return temp & (1 << bit);
 }
 
+static ALWAYS_INLINE void sys_set_bits(mem_addr_t addr, unsigned int mask)
```

Comment:
please document the new APIs in include/sys/sys_io.h

Reply:
I have added intel_socfpga_agilex_socdk.yaml and run twister but there are "Cmake build failure". Is it expected?

swiplab@PG15SWIPLAB1115:~/siclim/zephyrproject/zephyr$ ./scripts/twister -p intel_socfpga_agilex_socdk
ZEPHYR_BASE unset, using "/home/swiplab/siclim/zephyrproject/zephyr"
Install the anytree module to use the --test-tree option
Install tabulate python module with pip to use --device-testing option.
Install pyserial python module with pip to use --device-testing option.
Install tabulate python module with pip to use --device-testing option.
Install psutil python module with pip to run in Qemu.
can't import pykwalify; won't validate YAML (No module named 'pykwalify')
INFO    - Zephyr version: v2.7.0-rc2-144-g82c1c10d85d7
INFO    - JOBS: 12
INFO    - Using 'zephyr' toolchain.
INFO    - Building initial testcase list...
INFO    - 1466 test scenarios (1406 configurations) selected, 1017 configurations discarded due to filters.
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue
INFO    - Total complete:    3/ 389   0%  skipped: 1020, failed:    0ERROR   - Cmake build failure: /home/swiplab/siclim/zephyrproject/zephyr/samples/modules/nanopb for intel_socfpga_agilex_socdk

ERROR   - intel_socfpga_agilex_socdk samples/modules/nanopb/sample.modules.nanopb       FAILED: Cmake build failure
ERROR   - see: /home/swiplab/siclim/zephyrproject/zephyr/twister-out/intel_socfpga_agilex_socdk/samples/modules/nanopb/sample.modules.nanopb/build.log

Reply:
Not, this is not expected. you need to install the python dependencies, here pykwalify, look at the getting started guide

Reply:
Thanks, it can be built after install all the dependencies. There are 2 build error:

twister.log
----------
2021-10-02 04:36:33,849 - twister - ERROR - intel_socfpga_agilex_socdk tests/posix/fs/portability.posix.fs                [31mFAILED[39m: Build failure
2021-10-02 04:36:33,850 - twister - ERROR - see: [33m/home/swiplab/siclim/zephyrproject/zephyr/twister-out/intel_socfpga_agilex_socdk/tests/posix/fs/portability.posix.fs/build.log[39m
2021-10-02 04:36:33,925 - twister - ERROR - intel_socfpga_agilex_socdk tests/posix/fs/portability.posix.fs.newlib         [31mFAILED[39m: Build failure
2021-10-02 04:36:33,925 - twister - ERROR - see: [33m/home/swiplab/siclim/zephyrproject/zephyr/twister-out/intel_socfpga_agilex_socdk/tests/posix/fs/portability.posix.fs.newlib/build.log[39m

build.log
----------
-- Found BOARD.dts: /home/swiplab/siclim/zephyrproject/zephyr/boards/arm64/intel_socfpga_agilex_socdk/intel_socfpga_agilex_socdk.dts
intel_socfpga_agilex_socdk.dts.pre.tmp:28.21-39.5: Warning (simple_bus_reg): /soc/timer: missing or empty reg/ranges property
....
zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/build.make:103: recipe for target 'zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/fat_fs.c.obj' failed
make[2]: *** [zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/fat_fs.c.obj] Error 1
CMakeFiles/Makefile2:2796: recipe for target 'zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/all' failed
make[1]: *** [zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/all] Error 2
Makefile:90: recipe for target 'all' failed
make: *** [all] Error 2


I didn't see any error in the build.log, can you please advise is  warning 'missing or empty reg/ranges property' caused the build failed? I have tried to refer to other dts in Zephyr, I do see any difference on soc/timer when compare with others.

Reply:
good, you need to resolve those issues and make sure they are not coming with your change.

Reply:
I have fixed all warning caused by my changes. 

The twister is failed with the following error which is not caused by my changes. Am i good to resubmit my patches for review?

zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/build.make:103: recipe for target 'zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/fat_fs.c.obj' failed
make[2]: *** [zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/fat_fs.c.obj] Error 1
CMakeFiles/Makefile2:2798: recipe for target 'zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/all' failed
make[1]: *** [zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/all] Error 2
Makefile:90: recipe for target 'all' failed

Reply:
I have resubmitted the patches. All warnings coming from my changes are fixed.